soc/arm/nxp_imx/mimx8ml8_m7/soc.c

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (c) 2021, Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <fsl_clock.h>
+#include <fsl_common.h>
+#include <fsl_rdc.h>
+#include <init.h>
+#include <kernel.h>
+#include <soc.h>
+
+#include <dt-bindings/rdc/imx_rdc.h>
+#include "fsl_audiomix.h"
+
+/* Initialize MPU, configure memory attributes for each region */
+void SOC_InitMemory(void)
+{
+	/* __CACHE_REGION_START and __CACHE_REGION_SIZE are defined in the linker file */
+	extern uint32_t __CACHE_REGION_START[];
+	extern uint32_t __CACHE_REGION_SIZE[];
+	uint32_t cacheStart = (uint32_t)__CACHE_REGION_START;
+	uint32_t size       = (uint32_t)__CACHE_REGION_SIZE;
+	uint32_t i          = 0;
+
+	/* Disable I cache and D cache */
+	if (SCB_CCR_IC_Msk == (SCB_CCR_IC_Msk & SCB->CCR)) {
+		SCB_DisableICache();
+	}
+	if (SCB_CCR_DC_Msk == (SCB_CCR_DC_Msk & SCB->CCR)) {
+		SCB_DisableDCache();
+	}
+
+	/* Disable MPU */
+	ARM_MPU_Disable();
```

Comment:
MPU regions in other SOCs are defined in a separate file, as Zephyr has an architecture level MPU driver (in `arch` folder). See this file of an example of how custom MPU regions are defined: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/st_stm32/stm32h7/mpu_regions.c

Reply:
Thank you for the hint! See the new `mpu_regions.c` file.